Library/Source/NativeEngine.cpp

```diff
@@ -26,1152 +29,1151 @@ namespace bgfx
+    void SetIntArray4(const Napi::CallbackInfo& info);
+    void SetFloatArray(const Napi::CallbackInfo& info);
+    void SetFloatArray2(const Napi::CallbackInfo& info);
+    void SetFloatArray3(const Napi::CallbackInfo& info);
+    void SetFloatArray4(const Napi::CallbackInfo& info);
+    void SetMatrices(const Napi::CallbackInfo& info);
+    void SetMatrix3x3(const Napi::CallbackInfo& info);
+    void SetMatrix2x2(const Napi::CallbackInfo& info);
+    void SetFloat(const Napi::CallbackInfo& info);
+    void SetFloat2(const Napi::CallbackInfo& info);
+    void SetFloat3(const Napi::CallbackInfo& info);
+    void SetFloat4(const Napi::CallbackInfo& info);
+    void SetBool(const Napi::CallbackInfo& info);
+    Napi::Value CreateTexture(const Napi::CallbackInfo& info);
+    void LoadTexture(const Napi::CallbackInfo& info);
+    void LoadCubeTexture(const Napi::CallbackInfo& info);
+    Napi::Value GetTextureWidth(const Napi::CallbackInfo& info);
+    Napi::Value GetTextureHeight(const Napi::CallbackInfo& info);
+    void SetTextureSampling(const Napi::CallbackInfo& info);
+    void SetTextureWrapMode(const Napi::CallbackInfo& info);
+    void SetTextureAnisotropicLevel(const Napi::CallbackInfo& info);
+    void SetTexture(const Napi::CallbackInfo& info);
+    void DeleteTexture(const Napi::CallbackInfo& info);
+    void DrawIndexed(const Napi::CallbackInfo& info);
+    void Draw(const Napi::CallbackInfo& info);
+    void Clear(const Napi::CallbackInfo& info);
+    Napi::Value GetRenderWidth(const Napi::CallbackInfo& info);
+    Napi::Value GetRenderHeight(const Napi::CallbackInfo& info);
+
+    void DispatchAnimationFrameAsync(Napi::FunctionReference callback);
+
+    ShaderCompiler m_shaderCompiler;
+
+    ProgramData* m_currentProgram;
+
+    RuntimeImpl& m_runtimeImpl;
+
+    struct
+    {
+        uint32_t Width{};
+        uint32_t Height{};
+    } m_size;
+
+    bx::DefaultAllocator m_allocator;
+    uint64_t m_engineState;
+
+    // Scratch vector used for data alignment.
+    std::vector<float> m_scratch;
+};
+
+NativeEngine::Impl::Impl(void* nativeWindowPtr, RuntimeImpl& runtimeImpl)
+    : m_runtimeImpl{ runtimeImpl }
+    , m_currentProgram{ nullptr }
+    , m_size{ 1024, 768 }
+    , m_engineState{ BGFX_STATE_DEFAULT }
+{
+    bgfx::Init init{};
+    init.platformData.nwh = nativeWindowPtr;
+    bgfx::setPlatformData(init.platformData);
+
+    init.type = bgfx::RendererType::Direct3D11;
+    init.resolution.width = m_size.Width;
+    init.resolution.height = m_size.Height;
+    init.resolution.reset = BGFX_RESET_VSYNC;
+    bgfx::init(init);
+
+    bgfx::setViewClear(0, BGFX_CLEAR_COLOR | BGFX_CLEAR_DEPTH, 0x443355FF, 1.0f, 0);
+    bgfx::setViewRect(0, 0, 0, m_size.Width, m_size.Height);
+}
 
-        // Scratch vector used for data alignment.
-        std::vector<float> m_scratch;
-    };
+void NativeEngine::Impl::Initialize(Napi::Env& env)
+{
+    //EngineDefiner::Define(env, this); todo
```

Comment:
I don't understand errors with this one. Adding Definition:: before the function name gave me different error